subsystems/viamserver/viamserver.go

```diff
@@ -263,12 +302,14 @@ func (s *viamServer) Update(ctx context.Context, cfg *pb.DeviceSubsystemConfig,
 		s.logger.Info("awaiting user restart to run new viam-server version")
 		s.shouldRun = false
 	}
+	globalConfig = configFromProto(s.logger, cfg)
```

Comment:
Technically race-possible. (Shouldn't happen in normal operation, but a lot of fast-fails in a row or something might cause it to happen.) Needs lock or atomic like fast_start uses. Maybe https://pkg.go.dev/sync/atomic#Pointer ?

Reply:
done in 8c2219d